src/item.rs

```diff
@@ -31,13 +31,38 @@ impl ItemBundle {
         }
     }
 
-    pub fn spawn(mut commands: EntityCommands, item_spawn_meta: &ItemSpawnMeta) {
+    pub fn spawn(
+        mut commands: EntityCommands,
+        item_spawn_meta: &ItemSpawnMeta,
+        items_assets: &mut ResMut<Assets<ItemMeta>>,
+    ) {
         let ground_offset = Vec3::new(0.0, consts::GROUND_Y, consts::ITEM_LAYER);
         let transform_bundle = TransformBundle::from_transform(Transform::from_translation(
             item_spawn_meta.location + ground_offset,
         ));
 
         commands.insert_bundle(transform_bundle);
+
+        if let Some(item_meta) = items_assets.get_mut(&item_spawn_meta.item_handle) {
```

Comment:
This should actually be a panic if the asset isn't there, because this logic _needs_ to run for the item to spawn correctly.

We are already pre-loading all the assets before spawning the level, so the asset get shouldn't fail unless there is a bug we need to fix with the pre-loader, and we'll want to know about it in that case so panicking is still the right approach I think.